src/interval.cpp

```diff
@@ -64,4 +62,37 @@ bool Interval::operator<(const Interval& y) const
     return false;
 }
 
-bool Interval::empty() const { return length == 0; }
\ No newline at end of file
+bool Interval::empty() const { return length == 0; }
+
+void merge_intervals_within(std::vector<Interval>& intervals, const uint32_t dist)
+{
+    if (intervals.size() < 2) { // nothing to merge
+        return;
+    }
+    std::sort(intervals.begin(), intervals.end());
+
+    // assumption that iv1 < iv2
+    auto close { [dist](const Interval& iv1, const Interval& iv2) -> bool {
```

Comment:
This looks like an useful method to add to the `Interval` class itself, we might need to use it elsewhere